compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java

```diff
@@ -1264,6 +1264,10 @@ public void visit(BLangClassDefinition classDefinition) {
         lookupNodes(classDefinition.fields);
         lookupNodes(classDefinition.referencedFields);
         lookupNode(classDefinition.initFunction);
+        for (BLangFunction method : classDefinition.functions) {
```

Comment:
Isn't this a hack for method symbols not considering metadata location for its range? Seems like what's proposed in #38134 is the same fix.

Reply:
We are going to track theses changes in this PR. Will be closing the #38134

Reply:
We have done the same kind of thing for class-fields as well 
https://github.com/ballerina-platform/ballerina-lang/blob/32a3f8efa6b2cfa79ad4848fa25ced86923c3f0a/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java#L1260-L1262

This PR will address https://github.com/ballerina-platform/ballerina-lang/issues/37381 and https://github.com/ballerina-platform/ballerina-lang/issues/36351 but not https://github.com/ballerina-platform/ballerina-lang/issues/38053